File: Java/Medium/Accepted/shifting-letters_9.java
class Solution {
    public String shiftingLetters(String s, int[] shifts) {
        
        int n = s.length();
        char[] ch = s.toCharArray();
        int sum = 0;
        for(int i=shifts.length-1;i>=0;i--){
            sum = (sum+shifts[i])%26;
            ch[i] = (char)((ch[i]-'a'+sum)%26+'a');
        }
        return new String(ch);
    }
}

File: Rust/Medium/WrongAnswer/shifting-letters_9.rs
impl Solution {
    pub fn shifting_letters(s: String, shifts: Vec<i32>) -> String {
        
        let mut shifts = shifts;
        let mut sum = shifts.iter().sum::<i32>();
        let mut res = String::new();
        for i in 0..s.len() {
            let c = s.as_bytes()[i];
            let c = (c - b'a' + (sum % 26) as u8) % 26;
            res.push((c + b'a') as char);
            sum -= shifts[i];
        }
        res
    }
}